### PR TITLE
Improve shared missing scores

### LINKED ISSUE #1666: Shared Missing Scores List Bug
If a player for the shared list has an illegitimate maximum score, the respective variant appears in the missing scores list even if another player has a legitimate maximum score for it.

Example: 3-Player Rainbow (6 Suits) for Bonja97, melwen, and joelwool. Bonja97 has 30/30 with bottom-deck blind-play; melwen has the same score with no handicap options, but the variant still appears on the list.

https://hanab.live/scores/Bonja97
https://hanab.live/scores/melwen
https://hanab.live/shared-missing-scores/Bonja97/joelwool/melwen

### PR DESCRIPTION
**Pull request by timotree3**
Fixes #1666

### CODE REVIEW

File: server/src/http_shared_missing_scores.go [hunk 1]
```diff
@@ -56,9 +56,14 @@ func httpSharedMissingScores(c *gin.Context) {
 			continue
 		}
 		for j, variantStats := range variantStatsList {
-			for k, bestScore := range variantStats.BestScores {
-				if bestScore.Score > combinedVariantStatsList[j].BestScores[k].Score {
-					combinedVariantStatsList[j].BestScores[k] = bestScore
+			for k, candidateResult := range variantStats.BestScores {
+				workingBestResult := combinedVariantStatsList[j].BestScores[k]
+
+				if (candidateResult.Score > workingBestResult.Score) ||
```
Comment (Zamiell) [inline server/src/http_shared_missing_scores.go:62]: no parens here

File: server/src/http_shared_missing_scores.go [hunk 2]
```diff
@@ -56,9 +56,14 @@ func httpSharedMissingScores(c *gin.Context) {
 			continue
 		}
 		for j, variantStats := range variantStatsList {
-			for k, bestScore := range variantStats.BestScores {
-				if bestScore.Score > combinedVariantStatsList[j].BestScores[k].Score {
-					combinedVariantStatsList[j].BestScores[k] = bestScore
+			for k, candidateResult := range variantStats.BestScores {
```
Comment (Zamiell) [inline server/src/http_shared_missing_scores.go:59]: explain what the code does in a comment

Comment (timotree3) [reply server/src/http_shared_missing_scores.go:59]: done

Comment (Zamiell) [reply server/src/http_shared_missing_scores.go:59]: in addition to explaining what the code does, you should explain what the code *results in*.
like give an example where alice has A score with B mod and bob has C score with D mod.
what is the resulting behavior due to the algorithm you have chosen?

Comment (timotree3) [reply server/src/http_shared_missing_scores.go:59]: done

**Comment by timotree3** [comment]
The same logic was used in two different places and I made a function. Let me know if you want me to put the function in a different file.

**Comment by Zamiell** [comment]
excellent, thank you

**Pull request merged**